packages/create-subscription/README.md

```diff
@@ -0,0 +1,186 @@
+# create-subscription
+
+[Async-safe subscriptions are hard to get right.](https://gist.github.com/bvaughn/d569177d70b50b58bff69c3c4a5353f3)
+
+`create-subscription` provides an simple, async-safe interface to manage a subscription.
+
+## Who should use this?
+
+This utility is should be used for subscriptions to a single value that are typically only read in one place and may update frequently (e.g. a component that subscribes to a geolocation API to show a dot on a map).
+
+Other cases have better long-term solutions:
+* Redux/Flux stores should use the [context API](https://reactjs.org/docs/context.html) instead.
+* I/O subscriptions (e.g. notifications) that update infrequently should use [`simple-cache-provider`](https://github.com/facebook/react/blob/master/packages/simple-cache-provider/README.md) instead.
+* Complex libraries like Relay/Apollo should use this same technique (as referenced [here](https://gist.github.com/bvaughn/d569177d70b50b58bff69c3c4a5353f3)) in a way that is most optimized for their library usage.
+
+## What types of subscriptions can this support?
+
+This abstraction can handle a variety of subscription types, including:
+* Event dispatchers like `HTMLInputElement`.
+* Custom pub/sub components like Relay's `FragmentSpecResolver`.
+* Observable types like RxJS `BehaviorSubject` and `ReplaySubject`. (Types like RxJS `Subject` or `Observable` are not supported, because they provide no way to read the "current" value after it has been emitted.)
+* Native Promises.
+
+# Installation
+
+```sh
+# Yarn
+yarn add create-subscription
+
+# NPM
+npm install create-subscription --save
+```
+
+# Usage
+
+To configure a subscription, you must provide two methods: `getValue` and `subscribe`.
+
+```js
+import { createSubscription } from "create-subscription";
+
+const Subscription = createSubscription({
+  getValue(source) {
```

Comment:
For a while, I thought that this function was supposed to be `getInitialValue` until I read the implementation. Now I understand that you're handling this case.

1. Get the initial value for `state`
2. Wait to mount
3. Value changed while you were waiting to mount
4. Subscribe
5. The subscription isn't going to vend you the changed value, so you `getValue` while subscribing.

I'm worried people might do this, though I don't know what to do about it.

```
const Subscription = createSubscription({
  getValue(source) {
    return SomeConfig.initialValue;
  },
  subscribe(source, callback) {
    const {dispose} = source.listen(value => callback(value));
    return dispose;
  },
});

<Subscription source={someWebsocketOrSomething}>
  ...
</Subscription>
```

…where the source isn't something that offers a way to read the value synchronously.

Reply:
Any actionable suggestion to avoid this? Docs? Method names?